File: cloudstack/custom_client/listClusters.py
#!/usr/bin/python
import CloudStack

api = 'http://192.168.11.2:8080/client/api'
#admin
apikey='f0Q70X5oTpax-b7bvyzDQf2t_rKaK0rWkLXhejPRqI4i6IgiSq5rJ5_KXo7pCE9-HpYqAYllg_Td9675H3E31Q'
secret='8VxNvOkcKn7ibGAQhmI2L8frA7LmdN-IIGXNgmKJ_UxzWTpc487cFLE-vq0yLrXS6-dzL7luNecUyH0elrOA9g'

cs_object = CloudStack.Client(api, apikey, secret)
hosts = cs_object.listHosts()

#for key in hosts[0].keys():
#  print key + ',',
#print '\n'
# print key,type(host[key])

for host in hosts:
  if host.has_key('hypervisor'):
    for key in host.keys():
      if isinstance(host[key],unicode):
        print host[key].encode('utf-8') + ',',
      else:
        print str(host[key]) + ',',
    print

File: cloudstack/api/perl_client/cmds/listClusters.pl
#!/usr/bin/perl -w

use Net::CloudStack;
use JSON;

my $api = Net::CloudStack->new(
        base_url        => 'http://192.168.11.2:8080',
        api_path        => 'client/api?',
	#admin
        api_key         => 'f0Q70X5oTpax-b7bvyzDQf2t_rKaK0rWkLXhejPRqI4i6IgiSq5rJ5_KXo7pCE9-HpYqAYllg_Td9675H3E31Q',
        secret_key      => '8VxNvOkcKn7ibGAQhmI2L8frA7LmdN-IIGXNgmKJ_UxzWTpc487cFLE-vq0yLrXS6-dzL7luNecUyH0elrOA9g',
        xml_json        => 'json', #response format.you can select json or xml. xml is default.
        send_request    => 'yes',  #yes or no.
                                   #When you select yes,you can get response.
                                   #If you don't want to get response(only generating url),please input no.
    );

my $clusters = $api->proc("listClusters");
print $clusters;
